client/config.c

```diff
@@ -26,6 +28,51 @@ TDNFConfGetRpmVerbosity(
     return nLogLevel;
 }
 
+uint32_t
+TDNFParseOSInfo(
+    PTDNF_CONF pConf
+    )
+{
+    char *line = NULL;
+    uint32_t dwError = 0;
+    FILE *file = NULL;
+    file = fopen("/etc/os-release", "r");
+
+    if (!file) {
+        dwError = errno;
+        BAIL_ON_TDNF_SYSTEM_ERROR(dwError);
+    }
+
+    dwError = TDNFAllocateMemory(2*UserAgentHeaderMaxLength, sizeof(char), (void**)&line);
```

Comment:
You can use a temp buffer on the stack, then use `TDNFAllocate String()` to the the `pConf` member.